components/support/base/src/main/java/mozilla/components/support/base/observer/ObserverRegistry.kt

```diff
@@ -84,8 +84,12 @@ class ObserverRegistry<T> : Observable<T> {
             pausedObservers.remove(observer)
         }
 
+        // Unregister observers
         lifecycleObservers[observer]?.remove()
         viewObservers[observer]?.remove()
+        // Remove observers from map
+        lifecycleObservers.remove(observer)
+        viewObservers.remove(observer)
```

Comment:
I think we'd need to move all these 4 four calls up into the synchronized block as the method claims to be thread-safe, but that was also already a problem before and can be a follow-up (currently only invoked on the main thread afaik).

Reply:
Follow-up: https://github.com/mozilla-mobile/android-components/issues/3277